packages/twenty-front/src/modules/navigation/hooks/useSetLastVisitedViewForObjectMetadataNamePlural.ts

```diff
@@ -0,0 +1,63 @@
+import { lastVisitedObjectMetadataItemIdState } from '@/navigation/states/lastVisitedObjectMetadataItemIdState';
+import { lastVisitedViewPerObjectMetadataItemState } from '@/navigation/states/lastVisitedViewPerObjectMetadataItemState';
+import { objectMetadataItemsState } from '@/object-metadata/states/objectMetadataItemsState';
+import { useLazyPrefetchedData } from '@/prefetch/hooks/useLazyPrefetchData';
+import { PrefetchKey } from '@/prefetch/types/PrefetchKey';
+import { View } from '@/views/types/View';
+import { useRecoilCallback } from 'recoil';
+import { isDefined } from 'twenty-shared';
+
+export const useSetLastVisitedViewForObjectMetadataNamePlural = () => {
+  const { records: views, findManyRecords } = useLazyPrefetchedData<View>(
```

Comment:
introducing this hook but I think we should use a state and stop reading from apollo cache here